microovn/cmd/microovn/init.go

```diff
@@ -127,6 +162,23 @@ func (c *cmdInit) Run(cmd *cobra.Command, args []string) error {
 					return err
 				}
 
+				if customEncapsulationIpSupported {
+					// Register a potential custom encapsulation IP for other systems,
+					// so that when they will join the cluster, their encapsulation IP
+					// for the Geneve tunnel will be automatically configured.
+					key, encapIP, err := c.wantsCustomEncapsulationIP(tokenName)
+					if err != nil {
+						return err
+					}
+
+					if key != "" && encapIP != "" {
+						_, _, err = m.SQL(context.Background(), fmt.Sprintf("INSERT INTO config (key, value) VALUES ('%s', '%s')", key, encapIP))
```

Comment:
Sorry I missed this yesterday, but wouldn't it be cleaner to use one of the generated DB mapping functions (either `CreateConfigItem` or `UpdateConfigItem`) instead of raw SQL statememnt?

Reply:
We don't have access to the MicroCluster daemon's state here, we only have access to a client (hence the use of the `m.SQL` function that is forwarding the SQL statement to the daemon) and `CreateConfigItem` requireyou to have access to the instance of the daemon State in order to create the transaction with `CreateConfigItem` inside.

Reply:
Can this PR https://github.com/canonical/microcluster/pull/144 be useful?

Reply:
@mseralessandri I don't think this will help. Even with this PR, the state will remain internal to MicroCluster. To interact with the MicroCluster daemon holding the state, we can either:

* Define hooks that are passed to the MicroCluster app when its started.
* Can the MicroCluster API using a local client (what we do here)

As we need to have a hold on the state to create an SQL transaction in order to call `CreateConfigItem`, this is not doable (as of the current implementation of MicroCluster)

Reply:
I strongly disagree with using the `SQL` command like this, that should really only be used directly from the command line, it shouldn't be incorporated into other commands.

I think what you might need if you want to do this on the init node is an API endpoint to reach out to over the unix socket that takes a payload of config and applies it on the daemon using a transaction with the pre-defined SQL mappings.

But mainly, the thing that strikes me as most odd here is that you are making a change to the database with the "expectation" that those tokens will be used, but what happens if they are not? Will the data remain in the database? If that's the case, then the next time this same node tries to join, this block would fail because the key exists and we try to create it again, no?

I feel like this whole interactive step might be better to be supplied on the joiner side and the answers should be incorporated into map that is supplied to the `PreJoin` or `PostJoin` hooks.

Reply:
Also, does this introduce the risk of SQL injection? I see that query bindings are not being used.

Reply:
fixed